src/lib.rs

```diff
@@ -326,6 +356,73 @@ impl PrometheusMetricsBuilder {
     }
 }
 
+#[derive(Debug)]
+/// Configuration for a single metric
+///
+/// Allows configuring name and labels set for the metric
+pub struct ActixMetric {
+    name: String,
+    labels: Vec<String>,
+}
+
+impl ActixMetric {
+    /// Create a new metric configuration
+    pub fn new(name: &str, labels: Vec<&str>) -> ActixMetric {
+        ActixMetric {
+            name: name.to_string(),
+            labels: labels.into_iter().map(|s| s.to_string()).collect(),
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Configuration for the collected metrics
+///
+/// Stores individual metric configuration objects
+pub struct ActixMetricsConfiguration {
+    http_requests_total: ActixMetric,
+    http_requests_duration_seconds: ActixMetric,
+}
+
+impl ActixMetricsConfiguration {
+    /// Create the default metrics configuration
+    fn default() -> ActixMetricsConfiguration {
+        ActixMetricsConfiguration {
+            http_requests_total: ActixMetric::new(
+                "http_requests_total",
+                vec!["endpoint", "method", "status"],
+            ),
+            http_requests_duration_seconds: ActixMetric::new(
+                "http_requests_duration_seconds",
+                vec!["endpoint", "method", "status"],
+            ),
+        }
+    }
+
+    /// Create new metrics configuration with given input
+    pub fn new(
+        http_requests_total: ActixMetric,
+        http_requests_duration_seconds: ActixMetric,
+    ) -> ActixMetricsConfiguration {
+        ActixMetricsConfiguration {
+            http_requests_total,
+            http_requests_duration_seconds,
+        }
+    }
+
+    /// Set configs for http_requests_total metric
+    pub fn http_requests_total(mut self, value: ActixMetric) -> Self {
```

Comment:
I would prefer this also using the builder pattern names, i.e. `with_...`

Reply:
These setters are named similarly as in the rest of the codebase, for example `PrometheusMetricsBuilder`